### PR TITLE
GH-48 Boss bar cooking

### PR DESCRIPTION
**Pull request by Rollczi**
## Summary by CodeRabbit

- **New Features**
	- Updated Adventure API version to 4.18.0-SNAPSHOT.
	- Added support for boss bar notifications, including new commands and management features.
	- Introduced a new `TimerCommand` for timer functionalities.

- **Bug Fixes**
	- Enhanced serialization and deserialization processes for notices.

- **Tests**
	- Added new test cases for boss bar serialization and improved test readability.

- **Chores**
	- Updated imports and improved code readability across various components.

### CODE REVIEW

**Comment by Rollczi** [comment]
TODO: default values support (e.g. color, style)

**Review by vLuckyyy — APPROVED** [review]
LGTM

File: multification-core/src/com/eternalcode/multification/notice/resolver/bossbar/BossBarService.java [hunk 1]
```diff
@@ -0,0 +1,52 @@
...
+
+        Instant now = Instant.now();
+        Instant expiration = now.plus(duration);
+
+        if (content.progress().isEmpty()) {
+            updateProgress(expiration, duration, bossBar, viewer);
+        }
+
+        this.scheduler.schedule(() -> viewer.hideBossBar(bossBar), duration.toMillis(), TimeUnit.MILLISECONDS);
+    }
```
[the remark this answers is not in the document]
Comment (Rollczi) [reply multification-core/src/com/eternalcode/multification/notice/resolver/bossbar/BossBarService.java:36]: So stupid

File: multification-cdn/src/com/eternalcode/multification/cdn/MultificationNoticeCdnComposer.java [hunk 2]
```diff
@@ -219,6 +237,53 @@ private Result<Notice, Exception> deserializeSection(Section section) {
...
+    private Result<Blank, Exception> appendEntry(Entry entry, Notice.Builder builder) {
+        String value = this.deserializePiece(entry.getValue());
+        String key = entry.getName();
+
+        NoticeSerdesResult.Single result = new NoticeSerdesResult.Single(value);
+        Optional<NoticeDeserializeResult<?>> optional = this.noticeRegistry.deserialize(key, result);
+
+        if (optional.isEmpty()) {
+            return Result.error(new IllegalStateException("Unsupported notice type: " + key + ": " + value));
+        }
```
[the remark this answers is not in the document]
Comment (Rollczi) [reply multification-cdn/src/com/eternalcode/multification/cdn/MultificationNoticeCdnComposer.java:249]: Rel

**Review by igoyek — APPROVED** [review]

**Review by Hyd3r1 — APPROVED** [review]

**Review by P1otrulla — APPROVED** [review]

**Pull request merged**